includes/class-wcsatt-product.php

```diff
@@ -0,0 +1,856 @@
+						'regular_price' => WCS_ATT_Core_Compatibility::get_prop( $product, 'regular_price' )
+					);
+
+					$overridden_prices = WCS_ATT_Scheme_Prices::get_scheme_prices( $prices_array, $subscription_scheme );
+					$price             = $overridden_prices[ 'price' ];
+				}
+			}
+		}
+
+		return $price;
+	}
+
+	/**
+	 * Returns the regular price per subscription period.
+	 *
+	 * @param  WC_Product  $product     Product object.
+	 * @param  string      $scheme_key  Optional key to get the regular price of a specific scheme.
+	 * @param  string      $context     Function call context.
+	 * @return mixed                    The regular price charged per subscription period.
+	 */
+	public static function get_regular_price( $product, $scheme_key = '', $context = 'view' ) {
+
+		// In 'view' context, switch the active scheme if needed - and call 'WC_Product::get_regular_price'.
+		if ( 'view' === $context ) {
+
+			$active_scheme_key = self::get_subscription_scheme( $product );
+			$scheme_key        = '' === $scheme_key ? $active_scheme_key : $scheme_key;
+
+			// Attempt to switch scheme when requesting the price html of a scheme other than the active one.
+			$scheme_switch_required = $scheme_key !== $active_scheme_key;
+			$switched_scheme        = $scheme_switch_required ? self::set_subscription_scheme( $product, $scheme_key ) : false;
+
+			$regular_price = $product->get_regular_price();
+
+			// Switch back to the initially active scheme, if switched.
+			if ( $switched_scheme ) {
+				self::set_subscription_scheme( $product, $active_scheme_key );
+			}
+
+		// In 'edit' context, just grab the raw price from the scheme data + product props.
+		} else {
+
+			$subscription_scheme = self::get_subscription_scheme( $product, 'object', $scheme_key );
+			$regular_price       = WCS_ATT_Core_Compatibility::get_prop( $product, 'regular_price' );
+
+			if ( ! empty( $subscription_scheme ) ) {
+
+				if ( $subscription_scheme->has_price_filter() && apply_filters( 'wcsatt_price_filters_allowed', true, $product ) ) {
+
+					$prices_array = array(
+						'price'         => WCS_ATT_Core_Compatibility::get_prop( $product, 'price' ),
+						'sale_price'    => WCS_ATT_Core_Compatibility::get_prop( $product, 'sale_price' ),
+						'regular_price' => $regular_price
+					);
+
+					$overridden_prices = WCS_ATT_Scheme_Prices::get_scheme_prices( $prices_array, $subscription_scheme );
+					$regular_price     = $overridden_prices[ 'regular_price' ];
+				}
+			}
+		}
+
+		return $regular_price;
+	}
+
+	/**
+	 * Returns the sale price per subscription period.
+	 *
+	 * @param  WC_Product  $product     Product object.
+	 * @param  string      $scheme_key  Optional key to get the price of a specific scheme.
+	 * @param  string      $context     Function call context.
+	 * @return mixed                    The sale price charged per subscription period.
+	 */
+	public static function get_sale_price( $product, $scheme_key = '', $context = 'view' ) {
```

Comment:
👍 Opened https://github.com/Prospress/woocommerce-subscribe-all-the-things/issues/152